src/crypto.ts

```diff
@@ -0,0 +1,69 @@
+import blake from "blakejs";
+import { RegistryValue } from "./registry";
+import { FileID } from "./skydb";
+
+// NewHash returns a blake2b 256bit hasher.
+function NewHash() {
+  return blake.blake2bInit(32, null);
+}
+
+// HashAll takes all given arguments and hashes them.
+export function HashAll(...args: any[]): Uint8Array {
+  const h = NewHash();
+  for (let i = 0; i < args.length; i++) {
+    blake.blake2bUpdate(h, args[i]);
+  }
+  return blake.blake2bFinal(h);
+}
+
+// HashRegistryValue hashes the given registry value and returns it as a hex
+// encoded string
+export function HashRegistryValue(registryValue: RegistryValue): string {
+  return Buffer.from(
+    HashAll(
+      hexToUint8Array(registryValue.tweak),
+      encodeString(registryValue.data),
+      encodeNumber(registryValue.revision)
+    )
+  ).toString("hex");
+}
+
+// HashFileID hashes the given fileID and returns it as a hex encoded string
+export function HashFileID(fileID: FileID): string {
+  return Buffer.from(
+    HashAll(
+      encodeNumber(fileID.version),
+      encodeString(fileID.applicationID),
+      encodeNumber(fileID.fileType),
+      encodeString(fileID.filename)
+    )
+  ).toString("hex");
+}
+
+// stringToUint8Array converts a string to a uint8 array
+function stringToUint8Array(str: string): Uint8Array {
+  return Uint8Array.from(Buffer.from(str));
```

Comment:
`Buffer` doesn't work natively in the browser. You either need to use the string to uint8array function I have or use the `buffer` library, see my branch.